### ISSUE TITLE
Farcasterd panic in `swap_bob_funds_incorrect_amount` test

### ISSUE THREAD

**Issue by h4sh3d** [farcaster-project/farcaster-node#722]
Test is passing but `farcasterd` is crashing on message

```
thread 'main' panicked at 'attempt to subtract with overflow', src/farcasterd/runtime.rs:181:9
```

which correspond to line
```rust
    pub fn incr_funding_bitcoin_canceled(&mut self) {
        self.awaiting_funding_btc -= 1;
        self.funding_canceled_btc += 1;
    }

```

**Comment by h4sh3d** [comment]
Also happening in `swap_bob_maker_user_abort_sweep_btc` test

**Comment by h4sh3d** [comment]
In `swap_bob_maker_user_abort_sweep_btc` the problem is
1. btc funding is required (awaiting +1)
2. btc funding is completed (awaiting -1, funded btc +1)
3. sweep is requested
4. sweep is done, funding is canceled (awaiting -1, canceled +1)

We just did a double `-1` on awaiting, but the sequence is correct.

**Comment by h4sh3d** [comment]
> In `swap_bob_maker_user_abort_sweep_btc` the problem is
> 
>     1. btc funding is required (awaiting +1)
> 
>     2. btc funding is completed (awaiting -1, funded btc +1)
> 
>     3. sweep is requested
> 
>     4. sweep is done, funding is canceled (awaiting -1, canceled +1)
> 
> 
> We just did a double `-1` on awaiting, but the sequence is correct.

Fixed in #707 by introducing a set of swap ids for awaiting funding so double remove is handled properly.
